### PR TITLE
U bam collections dsdev 1898

### PR DESCRIPTION
**Pull request by ssaucedo**

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/vault/datamanagement/model/UBamCollection.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import scala.annotation.meta.field
+
+@ApiModel(value = "A collection of uBAMS")
+case class UBamCollection
+(
+  @(ApiModelProperty@field)(value = "The Vault IDs of the uBAMs included in this collection.", required = true)
+  members: Option[Seq[String]] = None,
+
+  @(ApiModelProperty@field)(value = "The metadata key-value pairs associated with this uBAM collection.", required = true)
+  metadata: Map[String, String],
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/vault/datamanagement/model/UBamCollection.scala]: Make this an `Option`

File: src/main/scala/org/broadinstitute/dsde/vault/datamanagement/domain/DataAccess.scala [hunk 2]
```diff
@@ -133,6 +133,44 @@ class DataAccess(val driver: JdbcProfile)
...
   }
 
+  private val membersForEntity = Compiled(
+    (entityGUID: Column[String]) => for {
+
+     //Retrieve the relation where our "entityGUID" parameter is the entity1GUID
+     relation <- relations
+     if relation.entity1GUID === entityGUID
+
+     // Using the foreign key, filter for the Relation.relation.entityType === INPUT_TYPE.entityType
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/vault/datamanagement/domain/DataAccess.scala]: Copy/Paste typo: `MEMBER_TYPE`

File: src/test/scala/org/broadinstitute/dsde/vault/datamanagement/services/UBamCollectionServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    "when accessing the /collections path" - {
+      //val members = Some(List("Ubam_id_1", "Ubam_id_2", "Ubam_id_3").toSeq)
+      val metadata = Map("key1" -> "someKey", "key2" -> "otherKey", "key3" -> "anotherKey")
+
+      val members = Option((
+        for (x <- 1 to 3) yield
+        DataManagementController.createUnmappedBAM(UnmappedBAM(Map.empty, Map.empty), "UBamCollectionServiceSpec").id.get
+        ).sorted.toSeq)
+
+      "POST should store a new Collection" in {
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/vault/datamanagement/services/UBamCollectionServiceSpec.scala]: Please add a test for the describe route.

**Comment by jmthibault79** [comment]
Looks good.  A few changes.  Please make these and rebase/squash.

@dags I was unsure who to assign this to, as multiple people worked on it.

File: src/test/scala/org/broadinstitute/dsde/vault/datamanagement/services/UBamCollectionServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import org.broadinstitute.dsde.vault.datamanagement.services.JsonImplicits._
+import spray.httpx.SprayJsonSupport._
+
+class UBamCollectionServiceSpec extends DataManagementDatabaseFreeSpec with UBamCollectionService {
+
+  def actorRefFactory = system
+
+  "UBamCollectionService" - {
+    "when accessing the /collections path" - {
+      //val members = Some(List("Ubam_id_1", "Ubam_id_2", "Ubam_id_3").toSeq)
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/vault/datamanagement/services/UBamCollectionServiceSpec.scala]: Remove comment

File: src/test/scala/org/broadinstitute/dsde/vault/datamanagement/services/UBamCollectionServiceSpec.scala [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          val collection = responseAs[UBamCollection]
+          collection.metadata should be(metadata)
+          collection.members should be(members)
+        }
+      }
+    }
+  }
+
+
+/*  override def afterAll(): Unit = {
```
Comment (jmthibault79) [inline src/test/scala/org/broadinstitute/dsde/vault/datamanagement/services/UBamCollectionServiceSpec.scala]: Remove comment

**Comment by ssaucedo** [comment]
Re-work is done , please make the re-review. Thanks

**Comment by kshakir** [comment]
@ssaucedo Please see #30 that adds versioning info. Before this branch is merged, start the versioning at `v1`, instead of no versioning. That way the `UBamCollectionService` doesn't have to use the `pathVersion()` directive used across the older services.

Also please rebase again off of develop. For any compile errors you run into, check the other services to see the updated syntax for directives.

Thanks!

**Comment by kshakir** [comment]
Please review my notes/patch in #33 regarding adding the properties and the tests.

If :+1:, run the normal merge process on #33 and it'll pull in changes here.

If :-1: or you'd like to make your own changes, that works too.

Either way please close the other PR after you're done and assign this PR back to me.

**Comment by kshakir** [comment]
Looks good! :+1: 

Please rebase onto develop, squash the commits, and then merge this PR.

**Pull request merged**